### PR TITLE
Fix logic for dns lookup.

### PR DESCRIPTION
**Pull request by pt247**
## Reference Issues or PRs
#2164 

## What does this implement/fix?
Fix logic for matching the domain IP. 

_Put a `x` in the boxes that apply_

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds a feature)
- [ ] Breaking change (fix or feature that would cause existing features not to work as expected)
- [ ] Documentation Update
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] Other (please describe):

## Testing

- [x] Did you test the pull request locally?
- [ ] Did you add new tests?

### CODE REVIEW

File: src/_nebari/stages/kubernetes_ingress/__init__.py [hunk 1]
```diff
@@ -67,23 +67,23 @@ def check_ingress_dns(stage_outputs: Dict[str, Dict[str, Any]], disable_prompt:
...
     def _attempt_dns_lookup(
-        domain_name, ip, num_attempts=NUM_ATTEMPTS, timeout=TIMEOUT
+        domain_name, ips, num_attempts=NUM_ATTEMPTS, timeout=TIMEOUT
     ):
         for i in range(num_attempts):
             try:
-                resolved_ip = socket.gethostbyname(domain_name)
-                if resolved_ip == ip:
+                _, _, resolved_ips = socket.gethostbyname_ex(domain_name)
+                if set(resolved_ips) == set(ips):
```
Comment (dcmcand) [inline src/_nebari/stages/kubernetes_ingress/__init__.py:79]: I think that I actually liked your original implementation better here. Even though a CNAME is better if your LB has multiple ips, by checking if the resolved ip is in the list of ips it leaves the possibility that someone could just do an A record for a single ip.

If we are going to use this approach where we check all the ips, then the error message recommend a CNAME in the case of multiple IPs.

**Review by dcmcand — CHANGES_REQUESTED** [review]

File: src/_nebari/stages/kubernetes_ingress/__init__.py (hunk 1 above)
Comment (pt247) [reply src/_nebari/stages/kubernetes_ingress/__init__.py:79]: Cool. I will change it back.

Comment (pt247) [reply src/_nebari/stages/kubernetes_ingress/__init__.py]: Done

**Review by dcmcand — APPROVED** [review]
@pt247 thanks for the PR. :rocket:

**Pull request merged**